admin/class-export.php

```diff
@@ -207,17 +206,68 @@ private function write_file() {
 	/**
 	 * Zips the settings ini file
 	 *
-	 * @return boolean unsigned
+	 * @return bool|null
 	 */
 	private function zip_file() {
+		$is_zip_created = $this->create_zip();
+
+		// This one isn't needed anymore.
+		$this->remove_settings_ini();
+
+		if ( ! $is_zip_created ) {
+			$this->error = __( 'Could not zip settings-file.', 'wordpress-seo' );
+
+			return false;
+		}
+
+		$this->download_zip( 'settings.zip' );
+		$this->remove_zip();
+
+		return true;
+	}
+
+	/**
+	 * Creates the zipfile and returns true if it created successful.
+	 *
+	 * @return bool
+	 */
+	private function create_zip() {
 		chdir( $this->dir['path'] );
 		$zip = new PclZip( './settings.zip' );
 		if ( 0 === $zip->create( './settings.ini' ) ) {
 			return false;
 		}
 
-		$this->export_zip_url = $this->dir['url'] . '/settings.zip';
+		return file_exists( 'settings.zip' );
+	}
 
-		return true;
+	/**
+	 * Downloads the zip file.
+	 *
+	 * @param string $filename The name of the zip file to download.
+	 */
+	private function download_zip( $filename ) {
+		ob_clean();
```

Comment:
This deserves a comment about why this is here.

Something like: `// Clean any content that has already been output. Even by other plugins or faulty php files.`